rest/src/main/java/io/syndesis/rest/v1/handler/integration/IntegrationSupportHandler.java

```diff
@@ -46,4 +63,86 @@ public IntegrationSupportHandler( ProjectGenerator projectConverter) {
         return projectConverter.generatePom(integration);
     }
 
+
+    @POST
+    @Path("/import")
+    public Response importIntegration(InputStream is) {
+        try (ZipInputStream zis = new ZipInputStream(is)) {
+            int imported = 0;
+            while (true) {
+                ZipEntry entry = zis.getNextEntry();
+                if( entry == null ) {
+                    break;
+                }
+                if (EXPORT_MODEL_FILE_NAME.equals(entry.getName())) {
+                    ModelData[] models = Json.mapper().readValue(new FilterInputStream(zis) {
+                        @Override
+                        public void close() throws IOException {
+                            // We want to avoid closing zis
+                        }
+                    }, ModelData[].class);
+
+                    imported += importModels(models);
+                }
+                zis.closeEntry();
+            }
+            if (imported==0) {
+                LOG.info("Could not import integration: No integration data model found.");
+                return Response.status(Response.Status.BAD_REQUEST).build();
+            }
+            return Response.status(Response.Status.NO_CONTENT).build();
+        } catch (IOException e) {
+            if (LOG.isInfoEnabled()) {
+                LOG.info("Could not import integration: " + e, e);
+            }
+            return Response.status(Response.Status.BAD_REQUEST).build();
+        }
+    }
+
+    public int importModels(ModelData... models) throws IOException {
+        int count = 0;
+        for (ModelData model : models) {
+            switch (model.getKind()) {
+                case Integration:
+
+                    Integration integration = (Integration) model.getData();
+                    // Do we need to create it?
+                    if (dataManager.fetch(Integration.class, integration.getId().get()) == null) {
+                        dataManager.create(integration);
+                    } else {
+                        dataManager.update(integration);
+                    }
+                    count ++;
+                    break;
+
+                case Connection:
+
+                    // We only create connections, never update.
+                    Connection connection = (Connection) model.getData();
+                    if (dataManager.fetch(Connection.class, connection.getId().get()) == null) {
```

Comment:
Is the id() some kind of symbolic name or a real (universal ?) unique id ? How it is is guaranteed that there is no clash with an already existing id (not coming from an import) ? Would it make sense to add some kind of unique prefix / namespace to imported connections to indicate their source ?

Reply:
Our id's are GUIDs, there should be no clashes so I think this is safe.
matching on the name of the object would be the other option, but is less safe since you could have conflicts, but might make more sense since folks are in control of that more.